pkg/tfshim/sdk-v2/internal/tf/generate.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2016-2023, Pulumi Corporation.
```

Comment:
I would prefer to transfer this over to linker tricks if possible, but we can do that at a later date.

Reply:
The question I have about linker tricks, is there a way to do this that does not infect downstream consumers of the lib to have to apply he same "go mod replace" in the consuming Go module? I was assuming that's not going to work, so we'd be infecting the downstream with more "go mod replace", but I can run a quick experiment to verify.

Reply:
Yes I'm pretty sure that linker tricks would impose `go mod replace` hell on our library consumers which is pretty unfortunate.

Reply:
There are different kinds of linker tricks. If we use one like our shamed providers, then we impose downstream costs. If we do some of the ([`go:linkname`](https://github.com/pulumi/pulumi/blob/4dc2943ae42f153872641140c48349b6f4681b87/sdk/go/pulumi/provider_linked.go#L48)) module dodging that pulumi/pulumi does, it doesn't have downstream impacts.

Reply:
Hm interesting, that wasn't part of my toolkit thanks!